### PR TITLE
Implement window operators in c++

### PR DESCRIPTION
**Pull request by achoum**
Bonus:
- Using the benchmark script, we see a x2000 speed-up.
- There is not allocated work memory anymore (only the output).

Benchmark:

Before
```
Running lag benchmark with N=300000...
Build schedule
Run 1 operators
    1 / 1: SIMPLE_MOVING_AVERAGE [7.26928 s]
Execution in 7.26963 s
```

After
```
Running lag benchmark with N=300000...
Build schedule
Run 1 operators
    1 / 1: SIMPLE_MOVING_AVERAGE [0.00371 s]
Execution in 0.00405 s
```

### CODE REVIEW

**Review by ianspektor — APPROVED** [review]
speechless with both the speedup and the code. thanks :)

File: temporian/core/operators/window/base.py [hunk 1]
```diff
@@ -73,9 +76,15 @@ def __init__(
...
 
-    def window_length(self):
+    def window_length(self) -> Duration:
         return self._window_length
 
+    def has_sampling(self) -> bool:
+        return self._has_sampling
+
+    def output_dtypes(self) -> List[dtype.DType]:
+        return self._output_dtypes
```
[the remark this answers is not in the document]
Comment (achoum) [reply temporian/core/operators/window/base.py:86]: Good point.
*Done

File: .vscode/settings.json
[the remark this answers is not in the document]
Comment (achoum) [reply .vscode/settings.json:1]: Good point.
Reverted the change.

**Comment by achoum** [comment]
Thanks! :)

**Pull request merged**